snapcraft/internal/meta.py

```diff
@@ -325,7 +325,9 @@ def _reformat_desktop(self, name, desktop_file):
                 'The specified desktop file {!r} is missing the '
                 '"Exec" key'.format(desktop_file))
         # XXX: do we want to allow more parameters for Exec?
-        exec_value = '{}.{} %U'.format(self._config_data['name'], name)
+        snap_name = self._config_data['name']
+        exec_value = '{} %U'.format(
+            name if name == snap_name else '{}.{}'.format(snap_name, name))
```

Comment:
maybe split the function in two, one to retrieve a valid desktop file, another to reformat and write or 3 and make a reformat and a separate write.

This could also be a small `class DesktopParser` or similar and moved out of Packaging (this would be the best)